tools/scripts/localstack/start.py

```
@@ -20,6 +39,108 @@
+def _get_container_ip(name: str) -> str | None:
+    """
+    Returns the container's first bridge-network IPv4 address, or None if it has
+    no IP yet.
+
+    :param name: The LocalStack container name.
+    :return: The container's IPv4 address on its Docker bridge network.
+    """
+    # Assigning the command to a variable dodges Ruff S607 (partial executable path),
+    # which only fires on a list literal passed directly to `subprocess.run`.
+    inspect_cmd = ["docker", "inspect", "-f", "{{json .NetworkSettings.Networks}}", name]
+    result = subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)